azure/services/networkinterfaces/networkinterfaces.go

```diff
@@ -116,7 +129,42 @@ func (s *Service) Delete(ctx context.Context) error {
 	return result
 }
 
-// IsManaged returns always returns true as CAPZ does not support BYO network interfaces.
+// isManaged returns true if the network interface has an owned tag with the cluster name as value,
+// meaning that the network interface's lifecycle is managed.
+func (s *Service) isManaged(ctx context.Context, spec azure.ResourceSpecGetter) (bool, error) {
+	_, _, done := tele.StartSpanWithLogger(ctx, "networkinterfaces.Service.IsManaged")
+	defer done()
+
+	if spec == nil {
+		return false, errors.New("cannot get network interface to check if it is managed: spec is nil")
+	}
+
+	if s.TagsGetter == nil {
+		return false, fmt.Errorf("tagsgetter is nil")
+	}
```

Comment:
Any specific reason on choosing `fmt.Errorf()` than `errors.New()` ?

Reply:
Just force of habit. Changed it now.